src/java/org/apache/cassandra/index/sai/metrics/ColumnQueryMetrics.java

```diff
@@ -21,15 +21,15 @@
 
 import com.codahale.metrics.Meter;
 import com.codahale.metrics.Timer;
-import org.apache.cassandra.index.sai.IndexContext;
+import org.apache.cassandra.index.sai.StorageAttachedIndex;
 
 import static org.apache.cassandra.metrics.CassandraMetricsRegistry.Metrics;
 
 public abstract class ColumnQueryMetrics extends AbstractMetrics
 {
-    protected ColumnQueryMetrics(IndexContext indexContext)
+    protected ColumnQueryMetrics(StorageAttachedIndex index)
```

Comment:
Here and in `IndexMetrics` would it make more sense to just use `IndexIdentifier` instead of having to pass the whole `StorageAttachedIndex`?

Reply:
It makes sense here but not in the `IndexMetrics`. That uses a number of methods on the index for metrics calculations.